datafusion/core/src/catalog/mod.rs

```diff
@@ -295,6 +296,118 @@ impl CatalogProvider for MemoryCatalogProvider {
     }
 }
 
+/// Resolve all table references in the SQL statement.
+///
+/// ## Example
+///
+/// ```
+/// use datafusion_sql::parser::DFParser;
+/// use datafusion::catalog::resolve_table_references;
+///
+/// let query = "SELECT a FROM foo where x IN (SELECT y FROM bar)";
+/// let statement = DFParser::parse_sql(query).unwrap().pop_back().unwrap();
+/// let table_refs = resolve_table_references(&statement, true).unwrap();
+/// assert_eq!(table_refs.len(), 2);
+/// assert_eq!(table_refs[0].to_string(), "foo");
```

Comment:
BTW this example fails for me locally (it seems "bar" is returned) -- perhaps related to the fact code is gathered in a hash table. Perhaps we could add a sort (or use a BTreeSet rather than a HashSet) to ensure the output order is consistent

Reply:
Nice catch, I've switched to a BTreeSet.